pkg/resources/external_oauth_integration.go

```diff
@@ -41,548 +46,777 @@ var oauthExternalIntegrationSchema = map[string]*schema.Schema{
+			externalOauthSnowflakeUserMappingAttribute, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "EXTERNAL_OAUTH_SNOWFLAKE_USER_MAPPING_ATTRIBUTE"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			externalOauthScopeDelimiter, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "EXTERNAL_OAUTH_SCOPE_DELIMITER"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			if err = handleExternalChangesToObjectInDescribe(d,
+				describeMapping{"external_oauth_issuer", "external_oauth_issuer", externalOauthIssuer.Value, externalOauthIssuer.Value, nil},
+				describeMapping{"external_oauth_jws_keys_url", "external_oauth_jws_keys_url", externalOauthJwsKeysUrl.Value, listValueToSlice(externalOauthJwsKeysUrl.Value, false), nil},
+				describeMapping{"external_oauth_any_role_mode", "external_oauth_any_role_mode", externalOauthAnyRoleMode.Value, externalOauthAnyRoleMode.Value, nil},
+				describeMapping{"external_oauth_rsa_public_key", "external_oauth_rsa_public_key", externalOauthRsaPublicKey.Value, externalOauthRsaPublicKey.Value, nil},
+				describeMapping{"external_oauth_rsa_public_key_2", "external_oauth_rsa_public_key_2", externalOauthRsaPublicKey2.Value, externalOauthRsaPublicKey2.Value, nil},
+				describeMapping{"external_oauth_blocked_roles_list", "external_oauth_blocked_roles_list", externalOauthBlockedRolesList.Value, listValueToSlice(externalOauthBlockedRolesList.Value, false), nil},
+				describeMapping{"external_oauth_allowed_roles_list", "external_oauth_allowed_roles_list", externalOauthAllowedRolesList.Value, listValueToSlice(externalOauthAllowedRolesList.Value, false), nil},
+				describeMapping{"external_oauth_audience_list", "external_oauth_audience_list", externalOauthAudienceList.Value, listValueToSlice(externalOauthAudienceList.Value, false), nil},
+				describeMapping{"external_oauth_token_user_mapping_claim", "external_oauth_token_user_mapping_claim", externalOauthTokenUserMappingClaim.Value, listValueToSlice(externalOauthTokenUserMappingClaim.Value, true), nil},
+				describeMapping{"external_oauth_snowflake_user_mapping_attribute", "external_oauth_snowflake_user_mapping_attribute", externalOauthSnowflakeUserMappingAttribute.Value, externalOauthSnowflakeUserMappingAttribute.Value, nil},
+				describeMapping{"external_oauth_scope_delimiter", "external_oauth_scope_delimiter", externalOauthScopeDelimiter.Value, externalOauthScopeDelimiter.Value, nil},
+			); err != nil {
+				return diag.FromErr(err)
+			}
 		}
-	}
-	if d.HasChange("rsa_public_key_2") {
-		val, ok := d.GetOk("rsa_public_key_2")
-		if ok {
-			alterInput.ExternalOauthRsaPublicKey2 = val.(string)
-			alterInput.ExternalOauthRsaPublicKey2Ok = true
-			runAlter = true
-		} else {
-			unsetInput.ExternalOauthRsaPublicKey2Ok = true
-			runUnset = true
+
+		if !d.GetRawConfig().IsNull() {
```

Comment:
use the new helper (i think you can find it in warehouse.go)

Reply:
Done.